cmd/sedad/utils/merkle.go

```diff
@@ -0,0 +1,107 @@
+package utils
+
+import (
+	"bytes"
+	"hash"
+	"math/bits"
+
+	"golang.org/x/crypto/sha3"
+)
+
+// RootFromEntries computes the root of the Merkle Hash Tree whose
+// leaves are the entries given in byte slices in the provided order.
+// It largely follows RFC-6962 with some modifications:
+//   - It uses SHA3-256 hashing algorithm.
+//   - Leaves are double-hashed for second preimage resistance. No hashing
+//     prefixes are used.
+//   - Each hash pair is sorted to make proofs more succinct.
+//   - "Super root" computation is supported.
+func RootFromEntries(entries [][]byte) []byte {
+	return rootFromEntries(sha3.New256(), entries)
```

Comment:
I think that this hashing algorithm is not the same as `keccak256`, which is widely used in blockchains.

Usually, keccak256 is the algorithm that won the NIST competition, but then during the SHA3 standarization they did minor changes that ended up to, what I think, it is the New256. Best is to check with some test vectors! :D

Reply:
Ah, didn't realize this. I see that geth uses legacy keccak256. Updated the hash function and added some unit tests.